apps/src/code-studio/progress.js

```diff
@@ -237,49 +227,29 @@ function initViewAs(store, scriptData) {
  * as appropriate
  */
 function queryUserProgress(store, scriptData, currentLevelId) {
-  const onOverviewPage = !currentLevelId;
-  const pageType = currentLevelId ? 'level' : 'script_overview';
-
-  $.ajax('/api/user_progress/' + scriptData.name, {
-    data: {
-      user_id: clientState.queryParams('user_id')
+  const userId = clientState.queryParams('user_id');
+  const onComplete = data => {
```

Comment:
Note: I created this `onComplete` callback (rather than moving its functionality into `progressRedux#queryUserProgress`) because the the actions called in it are not related to user progress or redux -- they are UI actions (e.g., `shoeDisabledBubblesModal()` and `renderTeacherPanel()`). Also, not all consumers of `progressRedux#queryUserProgress` will want to trigger these actions

Reply:
Great call! It looks like you've kept just the right stuff out of redux.